realm/realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -419,7 +428,7 @@ private void categorizeRequiredField(Element element, VariableElement variableEl
         if (Utils.isPrimitiveType(variableElement)) {
             Utils.error(String.format(
                     "@Required annotation is unnecessary for primitive field \"%s\".", element));
-        } else if (Utils.isRealmList(variableElement) || Utils.isRealmModel(variableElement)) {
+        } else if (Utils.isRealmInteger(variableElement) || Utils.isRealmModel(variableElement) || Utils.isRealmList(variableElement)) {
```

Comment:
I believe they should *always* be required and not nullable.  I was just having this discussion with @zaki50 .  I don't understand, at all, what it would mean for a counter to be null.